### ISSUE TITLE
Policy and baseline number of states assertion on ac and a2c

### ISSUE THREAD

**Issue by ThLind** [tensorforce/tensorforce#823]
Hi,
I upgraded to latest version yesterday from 0.6.2 and now I always get the following assertion error:
` assertion failed: [Policy and baseline cannot depend on a different number of previous states if baseline_optimizer is None.] [Condition x == y did not hold element-wise:] [x (policy/PartitionedCall_1:0) = ] [53] [y (baseline/PartitionedCall_1:0) = ] [0]
	 [[{{node agent/StatefulPartitionedCall/agent/cond_1/then/_320/agent/StatefulPartitionedCall/agent/assert_equal_1/Assert/AssertGuard/else/_839/agent/assert_equal_1/Assert/AssertGuard/Assert}}]] [Op:__inference_observe_9395]

Function call stack:
observe
`

This is my agent config:
`Agent.create(
            environment=environment,
            rnn=53,
            agent='ac',
            learning_rate=0.0001,
            batch_size=300,
            discount=0.49,
            l2_regularization=0.0001,
            exploration=0.3,
            huber_loss=0.1,
            target_update_weight=0.6,
            target_sync_frequency=8,
            horizon=16,
            use_beta_distribution=True
        )`

This is the environment states: {'type': 'float', 'shape': (21, 7), 'min_value': -10, 'max_value': 10}
and the actions: {'type': 'int', 'shape': (10, 7), 'num_values': 44}

I hope you can help me with this. 

Best Regards
Thilo

**Comment by ThLind** [comment]
Just found the solution. Sometimes it helps to just write down the settings outside of the IDE ;-)

Added critic network configuration with same rnn length solved it.

**Comment by AlexKuhnle** [comment]
Great -- yes, that's the problem here, currently not all combinations of RNN / reward horizon lengths are supported.
